src/test/java/org/mule/test/http/api/HttpRequestAttributesTestCase.java

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.http.api;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+
+import org.mule.extension.http.api.HttpRequestAttributes;
+import org.mule.runtime.api.util.MultiMap;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+import org.junit.Test;
+
+public class HttpRequestAttributesTestCase {
+
+  private static final String TO_STRING_COMPLETE = "org.mule.extension.http.api.HttpRequestAttributes\n" +
+      "{\n" +
+      "   Request path=/request/path\n" +
+      "   Method=GET\n" +
+      "   Listener path=/listener/path\n" +
+      "   Local Address=http://127.0.0.1:8080/\n" +
+      "   Query String=queryString\n" +
+      "   Relative Path=/relative/path\n" +
+      "   Remote Address=http://10.1.2.5:8080/\n" +
+      "   Request Uri=http://127.0.0.1/gateway\n" +
+      "   Scheme=scheme\n" +
+      "   Version=1.0\n" +
+      "   Headers=[\n" +
+      "      header2=header2\n" +
+      "      header1=header1\n" +
+      "   ]\n" +
+      "   Query Parameters=[\n" +
+      "      queryParam1=queryParam1\n" +
+      "      queryParam2=queryParam2\n" +
+      "   ]\n" +
+      "   URI Parameters=[\n" +
+      "      uriParam1=uriParam1\n" +
+      "      uriParam2=uriParam2\n" +
+      "   ]\n" +
+      "}";
+
+  private static final String TO_STRING_EMPTY = "org.mule.extension.http.api.HttpRequestAttributes\n" +
+      "{\n" +
+      "   Request path=null\n" +
+      "   Method=null\n" +
+      "   Listener path=null\n" +
+      "   Local Address=null\n" +
+      "   Query String=null\n" +
+      "   Relative Path=null\n" +
+      "   Remote Address=null\n" +
+      "   Request Uri=null\n" +
+      "   Scheme=null\n" +
+      "   Version=null\n" +
+      "   Headers=[\n" +
+      "   ]\n" +
+      "}";
+
+  private static final String TO_STRING_QUERY_PARAMS = "org.mule.extension.http.api.HttpRequestAttributes\n" +
```

Comment:
Why is it necessary to distinguish the query params and uri params cases from the complete case?